ccmbioinfo/lfs
# CARDS-1736: Add ACL restriction for clinics

[Jira link](https://phenotips.atlassian.net/browse/CARDS-1736)

Complete redo of #1055 , replacing that PR.

This PR adds an ACL restriction that only applies if the user belongs to a group specified by the Visit Information form.

**To test**:
1. Create a new clinic
2. Create a new Visit Information form, fill out the Clinic question with your newly created clinic and save
3. Create any number of new forms whose subject is the clinic
4. Under Administration->Groups, add `testuser` new user to both TrustedUser and the clinic group generated in step 1 (or create a new user if you're not in test runmode)
5. Verify that you can see the Visit Information form from step 2 and any forms from step 3 while signed in as `testuser` (password `testpassword`)
6. Try removing the user from the group to ensure that they can no longer access anything
7. Try creating a second new clinic and editing the `clinic` question in the visit information form, and making sure the permissions match.
8. Try having 100 forms and making sure that the restriction does not slow down the website performance too much.
